inspirehep/modules/theme/jinja2filters.py

```diff
@@ -668,3 +669,26 @@ def format_date(datetext):
             return date
         elif len(datestruct) == 1:
             return datestruct[0]
+
+
+@blueprint.app_template_filter()
+def is_external_link(url):
+    """Checks if given url starts with the following."""
+    return not url.endswith(('.pdf', '.png', '.jpg', '.jpeg'))
+
+
+@blueprint.app_template_filter()
+def is_institute(institute):
+    """Checks if given url starts with the following."""
+    if institute.lower() in ['kekscan', 'ads', 'cds', 'hal', 'msnet', 'msnet']:
+        return True
+    return False
+
+
+@blueprint.app_template_filter()
+def weblinks(description):
+    """Renames external links based on the description given."""
+    theme = INSPIRETheme()
```

Comment:
You do not need to reinstantiate the extension again as it is done upon loading of the entire application. Otherwise we would still always re-read the file upon constructing this object. To get the reference to it use the following:
```python
current_app.extensions.get('inspire-theme').weblinks.get(description)
```